ui/src/app/utils/runtime-utils.tsx

```diff
@@ -356,13 +401,19 @@ export const useRuntimeStatus = (currentWorkspaceNamespace, currentGoogleProject
 // all runtime state.
 export const useCustomRuntime = (currentWorkspaceNamespace):
     [{currentRuntime: Runtime, pendingRuntime: Runtime}, (runtime: Runtime) => void] => {
+
   const {runtime, workspaceNamespace} = useStore(runtimeStore);
+  const {persistentDisk} = useStore(diskStore);
   const runtimeOps = useStore(compoundRuntimeOpStore);
   const {pendingRuntime = null} = runtimeOps[currentWorkspaceNamespace] || {};
   const [requestedRuntime, setRequestedRuntime] = useState<Runtime>();
+  const enablePD = serverConfigStore.get().config.enablePersistentDisk;
 
   // Ensure that a runtime gets initialized, if it hasn't already been.
   useRuntime(currentWorkspaceNamespace);
+  if (enablePD) {
```

Comment:
hooks are not meant to be conditionally installed. I would always call the hook and then within that hook decide whether or not to populate the initial value depending on the feature flag